go/engine/puk_roll.go

```diff
@@ -152,6 +152,24 @@ func (e *PerUserKeyRoll) inner(ctx *Context) error {
 		len(pukBoxes), pukPrev != nil, gen)
 	libkb.AddPerUserKeyServerArg(payload, gen, pukBoxes, pukPrev)
 
+	ekLib := e.G().GetEKLib()
+	var myUserEKBox *keybase1.UserEkBoxed
+	var newUserEKMetadata *keybase1.UserEkMetadata
+	if ekLib != nil && ekLib.ShouldRun(ctx.NetContext) {
+		signingKey, err := pukSeed.DeriveSigningKey()
+		if err != nil {
+			return err
+		}
+		merkleRoot, err := e.G().GetMerkleClient().FetchRootFromServer(ctx.NetContext, libkb.EphemeralKeyMerkleFreshness)
```

Comment:
can you move the `DeriveSigningKey()`and `merkleRoot` fetching into the public method on `EKLib`? i try to expose internal methods that take `merkleRoot` in case we want the same root but never force a caller to fetch it since it is just repeated code

Reply:
also we want to add some error checking here..